components/images/lib.ts

```diff
@@ -0,0 +1,48 @@
+import { maxImageSize } from '@/config'
+import { type Image } from '@/lib/schema'
+
+export interface UploadImage extends Image {
+  file: Blob
+  progress: EventTarget
+}
+
+export function isUpload(upload: Image | UploadImage): upload is UploadImage {
+  return 'file' in upload
+}
+
+export async function createUpload(file: File): Promise<UploadImage | void> {
+  const bitmap = await createImageBitmap(file)
```

Comment:
The createImageBitmap function can be resource-intensive. Consider adding a check to ensure the file is a valid image before creating the bitmap.

```suggestion
export async function createUpload(file: File): Promise<UploadImage | void> {
if (!file.type.startsWith('image/')) {
console.error('File is not a valid image')
return
}
  const bitmap = await createImageBitmap(file)
```